pkg/products/monitoring/reconciler.go

```diff
@@ -776,3 +773,30 @@ func (r *Reconciler) reconcileSubscription(ctx context.Context, serverClient k8s
 	)
 
 }
+
+func (r *Reconciler) getPagerDutySecret(ctx context.Context, serverClient k8sclient.Client) (string, error) {
+
+	var secret string
+
+	pagerdutySecret := &corev1.Secret{}
+	err := serverClient.Get(ctx, types.NamespacedName{Name: r.installation.Spec.PagerDutySecret,
+		Namespace: r.installation.Namespace}, pagerdutySecret)
+
+	if err != nil {
+		return "", fmt.Errorf("could not obtain pagerduty credentials secret: %w", err)
+	}
+
+	if len(pagerdutySecret.Data["PAGERDUTY_KEY"]) != 0 {
+		secret = string(pagerdutySecret.Data["PAGERDUTY_KEY"])
+
```

Comment:
These trailing empty lines here and on L794 look a little odd to me. NBD though

Reply:
Agreed and fixed.I had to google what "NBD" meant :)